### DISCUSSION TITLE
Cannot use bot account for running tgcf in past mode

### DISCUSSION THREAD

**Discussion by tissole** [aahnik/tgcf#126] [Q&A] [answered]
I'm trying to run the script on Windows 10 past mode with a bot account but it gives me this error

```shell
"telethon.errors.rpcerrorlist.BotMethodInvalidError: The API access for bot users is restricted. The method you tried to invoke cannot be executed as a bot (caused by GetHistoryRequest)"
```

What is causing this and how can I fix it?

**Comment by ankit-sinha-18** [comment, marked as answer]
Bot login can't be used to iterate through the history of a chat. **Telegram does not allow that.**

 Log in using your phone number. (user account)

Bot login will work while using `tgcf` in live mode but not in past mode.

**Comment by tissole** [comment]
@aahnik Please update the Wiki to clearly say so. Thank you!

**Reply by aahnik** [reply]
yup, I will update it. please remind me, after one week if don't do.

**Comment by Nachos25** [comment]
Hi, Im similar error.
Unhandled exception on handle_admin_commands
Traceback (most recent call last):
  File "D:\lib\site-packages\telethon\client\updates.py", line 570, in _dispatch_update
    await callback(event)
  File "C:\Users\Asus\PycharmProjects\Work\main.py", line 88, in handle_admin_commands
    response = await client.get_messages(event.chat_id, limit=1)
  File "D:\lib\site-packages\telethon\client\messages.py", line 603, in get_messages
    return await it.collect()
  File "D:\lib\site-packages\telethon\requestiter.py", line 113, in collect
    async for message in self:
  File "D:\lib\site-packages\telethon\requestiter.py", line 74, in __anext__
    if await self._load_next_chunk():
  File "D:\lib\site-packages\telethon\client\messages.py", line 184, in _load_next_chunk
    r = await self.client(self.request)
  File "D:\lib\site-packages\telethon\client\users.py", line 30, in __call__
    return await self._call(self._sender, request, ordered=ordered)
  File "D:\lib\site-packages\telethon\client\users.py", line 87, in _call
    result = await future
telethon.errors.rpcerrorlist.BotMethodInvalidError: The API access for bot users is restricted. The method you tried to invoke cannot be executed as a bot (caused by GetHistoryRequest)
Pls, help me.

**Reply by MobCode100** [reply]
> Hi, Im similar error.
> Unhandled exception on handle_admin_commands
> Traceback (most recent call last):
>   File "D:\lib\site-packages\telethon\client\updates.py", line 570, in _dispatch_update
>     await callback(event)
>   File "C:\Users\Asus\PycharmProjects\Work\main.py", line 88, in handle_admin_commands
>     response = await client.get_messages(event.chat_id, limit=1)
>   File "D:\lib\site-packages\telethon\client\messages.py", line 603, in get_messages
>     return await it.collect()
>   File "D:\lib\site-packages\telethon\requestiter.py", line 113, in collect
>     async for message in self:
>   File "D:\lib\site-packages\telethon\requestiter.py", line 74, in __anext__
>     if await self._load_next_chunk():
>   File "D:\lib\site-packages\telethon\client\messages.py", line 184, in _load_next_chunk
>     r = await self.client(self.request)
>   File "D:\lib\site-packages\telethon\client\users.py", line 30, in __call__
>     return await self._call(self._sender, request, ordered=ordered)
>   File "D:\lib\site-packages\telethon\client\users.py", line 87, in _call
>     result = await future
> telethon.errors.rpcerrorlist.BotMethodInvalidError: The API access for bot users is restricted. The method you tried to invoke cannot be executed as a bot (caused by GetHistoryRequest)
> Pls, help me.

As noted by the owner, you can't use bot account in past mode, however if you wish to do that, you can use my plugin #478 which is still under pull request. This plugin allows you to change the sender to a bot account. My forked repo has it on the latest branch.

**Comment by Nachos25** [comment]
Can you help me whith install this plugin?

**Reply by ghost** [reply]
Login as a user. Bots have limitations.

**Reply by Nachos25** [reply]
Im tried it, dont help

**Reply by MobCode100** [reply]
Disclaimer: Using bot in past mode most probably will cause some issues, one is flood wait. Flood wait is known to have problems of messages being skipped. A higher delay for every forward can reduce the chance.

Well, if you still wanna go for it:
Download my [fork,](https://github.com/MobCode100/tgcf) branch latest, and unzip it. 
```
cd <downloaded_repo_path>
python3 -m venv .venv
source .venv/bin/activate
curl -sSL https://install.python-poetry.org | python3 -
poetry install
```

then try to run tgcf-web or tgcf. It should work.

**Reply by djfaizp** [reply]
your fork is showing 404

**Reply by MobCode100** [reply]
Because it has been merged in the tgcf's main branch, so just use this repo, as the replacement of my fork, use the commands above tho.

**Comment by Nachos25** [comment]
Can inline buttons help with this error?
